### PR TITLE
Rename `DataFramesDataset` to `PandasDataset`

### LINKED ISSUE #2056: Rename `DataFramesDataset` to `PandasDataset`
After some discussion with @lostella we want to 

- [x] rename `DataFramesDataset` to `PandasDataset`
- [x] support `pandas.Series` objects as a dataset with only a `target` column
- [x] treat a `DataFrame` with just one column like a `Series` object

As part of this we should also

- [x] fix naming used in the documentation notebook from #2051

### PR DESCRIPTION
**Pull request by rsnirwan**
resolves #2056

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by jaheba** [comment]
Would it make sense to rename `dataframe.py` to something else? `pandas.py` for example.

Also can you check that you change all references? I've updated `README.md` to use the dataset and it looks it's missing from your PR.

**Comment by rsnirwan** [comment]
ah yes .. I recursively checked in `src`, `test` and `doc` and that way missed the top level files. thanks!

File: test/dataset/test_pandas.py [hunk 1]
```diff
@@ -15,7 +15,7 @@
 import numpy as np
 import pytest
 
-from gluonts.dataset import dataframe
+from gluonts.dataset import pandas
```
Comment (jaheba) [inline test/dataset/test_pandas.py:18]: Now this looks a bit weird! Maybe we should rename this to `pandas_dataset` , or just `pd` -- oh wait!

Let's leave it like that.

**Review by jaheba — APPROVED** [review]

File: docs/index.rst [hunk 2]
```diff
@@ -41,7 +41,7 @@ See :doc:`community/devsetup` for setup instructions in case you want to develop
 
    tutorials/forecasting/quick_start_tutorial
    tutorials/forecasting/extended_tutorial
-   tutorials/data_manipulation/dataframesdataset
+   tutorials/data_manipulation/pandasdataframes
```
Comment (lostella) [inline docs/index.rst:44]: This did not need to change (since the tutorial file name did not change)

Comment (lostella) [reply docs/index.rst:44]: Wondering why the docs build did not fail?

Comment (jaheba) [reply docs/index.rst:44]: Because the file was actually renamed? ;)

Comment (lostella) [reply docs/index.rst:44]: Ok, didn't see that

**Pull request merged**